core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1644,6 +1644,13 @@ class KafkaApis(val requestChannel: RequestChannel,
         if (shareCoordinator.isEmpty) {
           return (Errors.INVALID_REQUEST, Node.noNode)
         }
+        try {
+          val _ = SharePartitionKey.getInstance(key)
```

Comment:
If we just need to validate if the key is correct then why don't we have a validate method and not create a instance on SharePartitionKey when it is not required?